### PR TITLE
Data processing

### PR DESCRIPTION
**Pull request by DoDucAnh2810**
Finished data processing. Have not test fully.

### CODE REVIEW

**Review by CodeNachos — APPROVED** [review]
dont wanna be annoying but where tf are the comments???

**Comment by TheAmazingSui** [comment]
Read your code. Had a stroke. Approved.

File: arm_data_processing.c [hunk 1]
```diff
@@ -27,12 +27,362 @@ Contact: Guillaume.Huard@imag.fr
...
+			shifter_operand = ror(Rm_value, get_bits(Rs_value, 4, 0));
+			shifter_carry_out = get_bit(Rm_value, get_bits(Rs_value, 4, 0) - 1);
+		}
+	} else
+		return UNDEFINED_INSTRUCTION;
+
+	/* CALCULATE RESULT BASED OPCODE */
+	switch (opcode) {
+	case AND:
+		result = Rn_value & shifter_operand;
```
Comment (TheAmazingSui) [inline arm_data_processing.c:214]: I skimmed through your code and the documentation, so I may be wrong, but shouldn't you check here whether rn (or rather before, when you extracted the value in bits) is equal to 15 and so you should add +8 to the address?
In the ARM documentation, it says:
Use of R15: If R15 is specified as register Rn, the value used is the address of the current instruction plus
eight.
Maybe you do this and I missed it...

Comment (CodeNachos) [reply arm_data_processing.c:214]: where did you saw this. I can only find when is with S, this case he treated

Comment (CodeNachos) [reply arm_data_processing.c:214]: ok found it. Yeah youssef is right i think there should be a case there to check if Rn is 15.

Comment (DoDucAnh2810) [reply arm_data_processing.c:214]: Aight I will add it

**Comment by DoDucAnh2810** [comment]
> Read your code. Had a stroke. Approved.

Tell me how to improve it. Other information is useless to me.

**Pull request merged**